### PR TITLE
fixed some typos.

### PR DESCRIPTION
**Pull request by dustinson**
A few things jumped out to me so I thought I'd help for the next reader.  Thx!

### CODE REVIEW

File: Docs/index.md [hunk 1]
```diff
@@ -2,18 +2,18 @@
 
 ## What is llbuild2?
 
-llbuild2 is a functional, artifact-based and extensible build system framework designed for large scale projects,
+llbuild2 is a functional, artifact-based, and extensible build system framework designed for large scale projects,
```
Comment (dmbryson) [inline Docs/index.md:5]: Ah the good ol' Oxford comma.    I approve.

File: Docs/index.md [hunk 2]
```diff
@@ -2,18 +2,18 @@
...
 
 ### Functional build systems
 
-Functional systems are caracterized by enforcing that the results (values) of evaluations (functions) are only affected
+Functional systems are characterized by enforcing that the results (values) of evaluations (functions) are only affected
 by the declared inputs (keys). Systems defined in such way can be smarter about the way evaluations are scheduled and
 cached. Some of the benefits of functional systems include:
 
-* If an evaluation is only affected by the input key, then the result can be memoized (cached) for future requests of
+* If an evaluation is only affected by the input key, then the result can be memorized (cached) for future requests of
```
Comment (dmbryson) [inline Docs/index.md:16]: I think memoized is actually more correct here.

**Comment by dmbryson** [comment]
Thanks for the corrections!

File: Docs/index.md [hunk 3]
```diff
@@ -40,7 +40,7 @@ llbuild2 is architected around the 2 phases that happen in any build system impl
 
 llbuild2 makes heavy use of CAS (Content Addressable Storage) technologies. With CAS, data, file and directory
 structures can be represented and accessed by the digest (or hash) of its contents. Using CAS identifiers, llbuild2 can
-detect when changes have ocurred to any portion of the build graph, and only reëvaluate the pieces that have never been
+detect when changes have occurred to any portion of the build graph, and only reevaluate the pieces that have never been
```
Comment (sergiocampama) [inline Docs/index.md:43]: I always have trouble with "occurred", I actually wrote it right the first time and changed it to this..

but I disagree on the ë change: https://www.newyorker.com/culture/culture-desk/the-curse-of-the-diaeresis

File: Docs/index.md [hunk 4]
```diff
@@ -2,14 +2,14 @@
...
-llbuild2 is a functional, artifact-based and extensible build system framework designed for large scale projects,
+llbuild2 is a functional, artifact-based, and extensible build system framework designed for large scale projects,
 focusing on maximizing the reusability of past evaluations. llbuild2 provides general abstractions related to build
 system problems and is designed with extensibility in mind. While llbuild2 by itself is not a build system in itself,
 you can build (hah) custom build systems with it.
 
 ### Functional build systems
 
-Functional systems are caracterized by enforcing that the results (values) of evaluations (functions) are only affected
+Functional systems are characterized by enforcing that the results (values) of evaluations (functions) are only affected
```
Comment (sergiocampama) [inline Docs/index.md:12]: my spanish heritage leaked :p

File: Docs/index.md (hunk 3 above)
Comment (dmbryson) [reply Docs/index.md:43]: TIL re: the diaeresis

**Pull request merged**